theta_dup/include/theta_sketch_dup.h

```diff
@@ -0,0 +1,2209 @@
+   * @param value uint16_t to be removed from the sketch
+   */
+  void remove(uint16_t value);
+
+  /**
+   * Remove one signed 16 bit integer from the theta-sketch.
+   * @param value int16_t to be removed from the sketch
+   */
+  void remove(int16_t value);
+
+  /**
+   * Remove one unsigned 8 bit integer from the theta-sketch.
+   * @param value uint8_t to be removed from the sketch
+   */
+  void remove(uint8_t value);
+
+  /**
+   * Remove one signed 8 bit integer from the theta-sketch.
+   * @param value int8_t to be removed from the sketch
+   */
+  void remove(int8_t value);
+
+  /**
+   * Remove one double-precision floating point value from the theta-sketch.
+   * @param value double to be removed from the sketch
+   */
+  void remove(double value);
+
+  /**
+   * Remove one single-precision floating point value from the theta-sketch.
+   * @param value float to be removed from the sketch
+   */
+  void remove(float value);
+
+  /**
+   * Remove one element of any type from this sketch.
+   * This is a "universal" remove that covers all cases above,
+   * but may produce different hashes.
+   * Be very careful to hash input values consistently using the same approach
+   * both over time and on different platforms
+   * and while passing sketches between C++ environment and Java environment.
+   * Otherwise two sketches that should represent overlapping sets will be
+   * disjoint For instance, for signed 32-bit values call update(int32_t) method
+   * above, which does widening conversion to int64_t, if compatibility with
+   * Java is expected
+   * @param data pointer to the data
+   * @param length of the data in bytes
+   */
+  void remove(const void* data, unsigned length);
+
+  /**
+   * Remove retained entries in excess of the nominal size k (if any)
+   */
+  void trim();
+
+  /**
+   * Converts this sketch to a compact sketch (ordered or unordered).
+   * @param ordered optional flag to specify if ordered sketch should be
+   * produced
+   * @return compact sketch
+   */
+  compact_theta_sketch_dup_alloc<A> compact(bool ordered = true) const;
+
+  virtual typename theta_sketch_dup_alloc<A>::const_iterator begin() const;
+  virtual typename theta_sketch_dup_alloc<A>::const_iterator end() const;
+
+  /**
+   * This method deserializes a sketch from a given stream.
+   * @param is input stream
+   * @param seed the seed for the hash function that was used to create the
+   * sketch
+   * @return an instance of a sketch
+   */
+  static update_theta_sketch_dup_alloc<A> deserialize(
+      std::istream& is, uint64_t seed = DEFAULT_SEED);
+
+  /**
+   * This method deserializes a sketch from a protobuf
+   * @param pb input stream
+   * @param seed the seed for the hash function that was used to create the
+   * sketch
+   * @return an instance of a sketch
+   */
+  static update_theta_sketch_dup_alloc<A> deserialize(
+      datasketches_pb::Update_theta_sketch_dup& pb,
+      uint64_t seed = DEFAULT_SEED);
```

Comment:
Declaration of the deserialization of the **derived class** update_theta_sketch_dup_alloc.
pb: variable of the protobuf generated class.